subprojects/launcher/src/main/java/org/gradle/launcher/cli/CommandLineActionFactory.java

```diff
@@ -184,20 +195,30 @@ public void run() {
         private final List<String> args;
         private final LoggingConfiguration loggingConfiguration;
         private final Action<ExecutionListener> action;
+        private final BuildLayoutParametersBuildOptionFactory buildLayoutParametersBuildOptionFactory;
+        private final StartParameterBuildOptionFactory startParameterBuildOptionFactory;
+        private final ParallelismBuildOptionFactory parallelismBuildOptionFactory;
+        private final DaemonBuildOptionFactory daemonBuildOptionFactory;
+        private final LoggingConfigurationBuildOptionFactory loggingConfigurationBuildOptionFactory;
 
-        WithLogging(ServiceRegistry loggingServices, List<String> args, LoggingConfiguration loggingConfiguration, Action<ExecutionListener> action) {
+        WithLogging(ServiceRegistry loggingServices, List<String> args, LoggingConfiguration loggingConfiguration, Action<ExecutionListener> action, BuildLayoutParametersBuildOptionFactory buildLayoutParametersBuildOptionFactory, StartParameterBuildOptionFactory startParameterBuildOptionFactory, ParallelismBuildOptionFactory parallelismBuildOptionFactory, DaemonBuildOptionFactory daemonBuildOptionFactory, LoggingConfigurationBuildOptionFactory loggingConfigurationBuildOptionFactory) {
             this.loggingServices = loggingServices;
             this.args = args;
             this.loggingConfiguration = loggingConfiguration;
             this.action = action;
+            this.buildLayoutParametersBuildOptionFactory = buildLayoutParametersBuildOptionFactory;
+            this.startParameterBuildOptionFactory = startParameterBuildOptionFactory;
+            this.parallelismBuildOptionFactory = parallelismBuildOptionFactory;
+            this.daemonBuildOptionFactory = daemonBuildOptionFactory;
+            this.loggingConfigurationBuildOptionFactory = loggingConfigurationBuildOptionFactory;
         }
 
         public void execute(ExecutionListener executionListener) {
-            CommandLineConverter<LoggingConfiguration> loggingConfigurationConverter = new LoggingCommandLineConverter();
-            CommandLineConverter<BuildLayoutParameters> buildLayoutConverter = new LayoutCommandLineConverter();
-            CommandLineConverter<ParallelismConfiguration> parallelConverter = new ParallelismConfigurationCommandLineConverter();
+            CommandLineConverter<LoggingConfiguration> loggingConfigurationConverter = new LoggingCommandLineConverter(loggingConfigurationBuildOptionFactory);
```

Comment:
You might wanna register these converters as services too and inject a `Set` of them, as described above. Then you can just do a for-loop over them instead of mentioning each individually.